nixos/modules/services/logging/pushlog.nix

```diff
@@ -0,0 +1,171 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+with lib;
+let
+  cfg = config.services.pushlog;
+  unitType = types.submodule {
+    options = {
+      match = mkOption {
+        type = types.str;
+        default = ".*";
+      };
+      priorities = mkOption {
+        type = types.listOf types.int;
+        default = [
+          0
+          1
+          2
+          3
+          4
+          5
+          6
+        ];
+      };
+      include = mkOption {
+        type = types.listOf types.str;
+        default = [ ];
+      };
+      exclude = mkOption {
+        type = types.listOf types.str;
+        default = [ ];
+      };
+    };
+  };
+in
+{
+  options.services.pushlog = {
+    enable = mkEnableOption "Enable pushlog service to forward journald logs to Pushover";
+
+    package = mkPackageOption pkgs "pushlog" { };
+
+    environmentFile = mkOption {
+      type = with types; nullOr str;
+      description = lib.mdDoc ''
+        File containing the Pushover API credentials, in the
+        format of an EnvironmentFile as described by systemd.exec(5)
+
+        PUSHLOG_PUSHOVER_TOKEN, PUSHLOG_PUSHOVER_USER_KEY
+      '';
+      default = null;
+    };
+
+    settings = {
```

Comment:
If this is supposed to be fed directly into an arbitrary settings file, then it's better to make this into an option with a submodule type, whose freeformType is equal to the type of a YAML document. This way Nix can safely ignore any unrecognized keys while still doing rudimentary type checking against invalid values

```nix
    settings = lib.mkOption {
      type = lib.types.submodule {
        freeformType = format.type;

        options = {
          # ...
        };
      };
    };
```